src/main/java/com/amazon/opendistroforelasticsearch/sql/rewriter/matchtoterm/TermFieldRewriter.java

```diff
@@ -235,7 +235,11 @@ private boolean isValidIdentifier(SQLObject expr) {
         SQLObject parent = expr.getParent();
         return
                 (parent instanceof SQLBinaryOpExpr
-                        && ((SQLBinaryOpExpr) parent).getOperator() == SQLBinaryOperator.Equality
+                 && (
```

Comment:
I commented earlier on similar case - those logical conditions are very long. Consider extracting this complex condition into named method

Reply:
Tried some refactoring.